python/tskit/tables.py

```diff
@@ -3020,6 +3026,99 @@ def union(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
 
+    def time_join(self, recent, time, continue_nodes=None):
+        ancient_ts = self.tree_sequence()
```

Comment:
We're not actually using ``ancient_ts`` here for anything, and creating a ``tree_sequence`` is quite expensive. We can get the sample nodes directly from the tables.